### ISSUE TITLE
dcc.Upload - need to explain that this is primarly for uploading files that aren't on the server

### ISSUE THREAD

**Issue by chriddyp** [plotly/dash-docs#335]
many folks in the community (https://community.plot.ly/t/uploading-mdf-files/17004) are using a `dcc.Upload` component as a way to get a filename of a file and then `file.read()` that file. this isn't really the use case for `dcc.Upload`: `dcc.Upload` is primarily for files that _aren't_ on the actual server (the app is deployed on a different machine than what the user has visited). 

If users just need to get the filename of a file to `file.read()` it, they might be better off using a `dcc.Dropdown` with options that are filenames (https://community.plot.ly/t/get-file-path-from-upload-component/9934/4)

**Comment by PleaseStateTheNatureOfYourInquiry** [comment]
Yes, this is true! But the problem is that in many cases there are hundreds of files to navigate. In my case, I want the flexibility of a file choser window, so that I can browse my computer to find and open a file and know the full path to it. I did find a work-around with dropdowns, but still, not ideal.
